### PR TITLE
Fix #1004 Auto-generate PHPDoc for Illuminate\Support\Carbon

### LINKED ISSUE #1004: Model when use Carbon, auto generate docblock not pass by phpstan.
this is my User Model diff after run `php artisan ide-helper:models -WR`

image

it will remove `\Carbon\Carbon` in docblock, would lead to result by phpstan: (https://github.com/nunomaduro/larastan)

```
 ------ -----------------------------------------------------------------------
  Line   User.php
 ------ -----------------------------------------------------------------------
  385    Property App\User::$vip_end (Illuminate\Support\Carbon|null) does not
         accept Carbon\Carbon.
  388    Property App\User::$vip_end (Illuminate\Support\Carbon|null) does not
         accept Carbon\Carbon.
  394    Property App\User::$vip_end (Illuminate\Support\Carbon|null) does not
         accept Carbon\Carbon.
 ------ -----------------------------------------------------------------------
```

Is there something I'm doing wrong? Please advise, thank you very much!

### PR DESCRIPTION
**Pull request by kylekatarnls**
Fix #1004 Auto-generate PHPDoc for Illuminate\Support\Carbon from Carbon\Carbon

### CODE REVIEW

**Comment by kylekatarnls** [comment]
This could help PHPStan and IDE to understand that `Illuminate\Support\Carbon` return `Illuminate\Support\Carbon` instances for methods where the `Carbon\Carbon` PHPDoc mention `Carbon\Carbon` return hints in `@method` annotations.

File: src/Generator.php [hunk 1]
```diff
@@ -82,6 +84,7 @@ public function generatePhpHelper()
     {
         $app = app();
         return $this->view->make('helper')
+            ->with('carbonPhpDoc', $this->getCarbonPhpDoc())
```
Comment (kylekatarnls) [inline src/Generator.php:87]: Should this variable be in snake_case?

**Comment by kylekatarnls** [comment]
I tested it with the following:
```php
/**
 * @property \Illuminate\Support\Carbon|null $last_action_at
 */
class User
{
  public function foo()
  {
    $this->last_action_at = now()->addDays(5);
  }
}
```
And this still throw the PHPStan error because both the `Illuminate\Support\Carbon` and the `Carbon\Carbon` PHPDoc are considered and so the return of addDays() is saw as `Illuminate\Support\Carbon | Carbon\Carbon` (the same happens for the IDE auto-completion).

**Comment by mfn** [comment]
@kylekatarnls I think we can close this one with the outcome of https://github.com/barryvdh/laravel-ide-helper/issues/1004 ?

**Comment by kylekatarnls** [comment]
Yep

**Pull request closed without merging**